cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -150,7 +153,12 @@ func handleLogLockEvent(
 	}
 
 	// Initiate the relay
-	return txs.RelayLockToCosmos(chainID, cdc, validatorAddress, validatorName, passphrase, &prophecyClaim, rpcURL)
+	err = txs.RelayLockToCosmos(chainID, cdc, validatorAddress, validatorName, passphrase, &prophecyClaim, rpcURL)
```

Comment:
Looks like we are creating a new cliCtx in RelayLockToCosmos each time. Should we be reusing the cliCtx across transactions instead of creating new ones? I don't recall exactly, but think the crash could have been due to this too rather than due to accessing the passphrase/key multiple times (not sure if we were even doing that)?

Also, what about Relaying burns? Does this function handle that too? It should be renamed just RelayProphecyToCosmos then?

Reply:
Good point, I'll add a utility function for this called on Relayer init. Related - Ethereum's tx context is currently transactOptsAuth, but as it requires the validator's current nonce, I think that one should remain at the transactional level.

Reply:
Attempting to start a second Ethereum Relayer would still hit the db resource error, even without witnessing a LogLock event. Likely both were errors that surfaced in different contexts.

Reply:
I made a note of this conversation in the new PR.